File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    day = defaultdict(list)

    for name, typ in clothes:
        day[typ].append(name)

    # print(day)
        
    for name, typ in day.items():
        # print(name, typ)
        tmp = len(typ) +1
        answer *= tmp

    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String, Integer> clothes_lst = new HashMap<>();
        for (String[] cloth: clothes) {
            clothes_lst.put(cloth[1], clothes_lst.getOrDefault(cloth[1], 0)+1);
        }
        
        System.out.println(clothes_lst);
        
        for (String key: clothes_lst.keySet()) {
            int cnt = clothes_lst.getOrDefault(key, 0);
            answer*=cnt+1;
        }
        
        return answer-1;
    }
}